### ISSUE TITLE
VkLayerTest.CopyImageTypeExtentMismatch dies if KHR_maintenance1 is not supported

### ISSUE THREAD

**Issue by chrisforbes** [KhronosGroup/Vulkan-LoaderAndValidationLayers#1812]
Like so:

```
$ build/tests/vk_layer_validation_tests --gtest_filter='*CopyImageTypeExtentMismatch'
Note: Google Test filter = *CopyImageTypeExtentMismatch
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from VkLayerTest
[ RUN      ] VkLayerTest.CopyImageTypeExtentMismatch
Unexpected: loader_validate_device_extensions: Extension 0 not found in enabled layer list extensions.
Unexpected: vkCreateDevice:  Failed to validate extensions in list
../tests/vktestbinding.cpp:295: Failure
Failed
Assertion: `vkCreateDevice(phy_.handle(), &info, __null, &dev) == VK_SUCCESS'
Segmentation fault (core dumped)
```

Blindly including the extension string causes device creation to fail, and things go poorly after that.

**Comment by chrisforbes** [comment]
Either:
- create another device for the test guts, which tries to enable the extension iff we've established that it exists, (preferred?) or

- Split the KHR_maintenance1 cases into a separate test where they can be skipped together if the extension isn't supported.

**Comment by chrisforbes** [comment]
Possibly ties into #1827 a bit

**Comment by daveh-lunarg** [comment]
The intention here was as Chris' first bullet: check for existence of KHR_maintenance1, enable it if offered, and modify the test behavior to match.  This test fails on our Android farm just as Chris listed above. I talked to Cody before pushing the PR, and thought I understood that the test was passing on Google's farm, so assumed it was a local issue here. The check/enable code comes verbatim from the positive test Maintenance1Tests() - which I presume is failing the same way?

In any case the KHRm1 extension is not essential to that test, only there to include a couple new VUs that I've proposed should be removed from the spec as redundant (although feedback is that they're more likely to remove the original overlapping VU and keep these). I can solve this particular issue by stripping it out of the test.

I actually like Chris' second bullet idea, move all KHRm1 awareness into a segregated separate set of tests (better yet, in another file).

**Comment by chrisforbes** [comment]
`Maintenance1Tests` worked because it skipped the test completely before trying to use the device it failed to create.

**Comment by cnorthrop** [comment]
The test is failing on Ubuntu using mesa, not Android.  LunarG should probably hook that up.  Paging @Tony-LunarG.

**Comment by daveh-lunarg** [comment]
Ok.  It's still not clear to me why it works in Maintenance1, because the CreateDevice() in InitFramework() happens the same way in both.  But, it's not integral to this test so I'm just going to remove it for now to close this issue.

And do some more thinking about how to deal with extension-specific testing...

**Comment by daveh-lunarg** [comment]
Should be resolved with PR #1831.

**Comment by cnorthrop** [comment]
Yes, the test passes on mesa now.  Although it crashes in the driver later in VkPositiveLayerTest.ClearDepthStencilWithValidRange, which @chrisforbes has already debugged.
